File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        
        // vector<int> result;
        // for(auto it = nums.begin(); it != nums.end(); ++it){
        //     auto it2 = find(it+1 ,  nums.end() , target - *it) ;
        //     if(it2 != nums.end()){
        //         result.push_back(it - nums.begin());
        //         result.push_back(it2 - nums.begin());
        //         break;
        //     }
        // }
        //  return result;
        
        unordered_map<int, int> _map;
         for( int i = 0; i < nums.size() ; ++i){
             int num = nums[i];
             int complement= target - num;
             auto it = _map.find(complement);
             if(it != _map.end()){
                 //found
                 return { it->second , i};
             }
             _map[num] = i;
         }
        return {};
    }
   
};

File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] arr = new int[2];
        int pos =0;
        HashMap<Integer,Integer>  hm = new HashMap<Integer,Integer>();        
        for(int i =0 ; i < nums.length ; i++){
            if(hm.containsKey(target - nums[i])){
                arr[pos++] = hm.get(target - nums[i]);
                arr[pos] = i;
                return arr;
            }else{
                hm.put(nums[i],i);
            }
        }
        return arr;
    }
}